plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -458,6 +458,35 @@ func ClusterRoles() []rbac.ClusterRole {
 				eventsRule(),
 			},
 		},
+		{
+			// a role for csi external components {csi-provisioner,csi-attacher,driver-registrar}
+			ObjectMeta: metav1.ObjectMeta{Name: "system:csi-external-provisioner"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("create", "delete", "list", "watch").Groups(legacyGroup).Resources("persistentvolumes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumeclaims").RuleOrDie(),
+				rbac.NewRule("get").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
```

Comment:
provisioners don't have a general need to get every secret in the cluster. a provisioner might:
* need no secret-related permissions
* need permissions to view a particular secret for provisioning
* need permissions to view secrets in a particular namespace for provisioning
* need permissions to create attach/mount-related secrets in a particular namespace

rather than grant broad permissions here on the off chance the provisioner needs them, I'd expect specific permissions to be granted as a particular provisioner or storage class requires.

Reply:
Ok, I will remove the secrets for now. We will have to implement some sort of workaround to allow for creating `Role`s in the test framework for GKE once we start adding tests for external CSI components that require secrets.

Reply:
> once we start adding tests for external CSI components that require secrets.

I'd actually expect those to come with recommended setups that put them in a namespace, set up a storage class for them, and granted them appropriate secret permissions within that namespace. Basically, the standard role (which you are defining here) takes care of the "every provisioner will have to do X"  things, and they get to grant any additional permissions they want in a manifest specific to their component.

Reply:
Right, when I remove the secrets it the roles will be consistent with "every provisioner will have to do X". The recommended setups will have the correct roles to grant whatever additional permissions they need. The only issue is that the test framework on GKE does not allow creation of `Roles` by default so there would just need to be a workaround in the test or a larger framework/GKE change would need to be done